localstack/services/apigateway/apigateway_listener.py

```diff
@@ -313,6 +315,60 @@ def return_response(self, method, path, data, headers, response):
             )
 
 
+class RequestValidator:
+    __slots__ = ["context", "apigateway_client"]
+
+    def __init__(self, context: ApiInvocationContext, apigateway_client):
+        self.context = context
+        self.apigateway_client = apigateway_client
+
+    def is_request_valid(self) -> bool:
+        # make all the positive checks first
+        if self.context.resource is None or "resourceMethods" not in self.context.resource:
+            return True
+
+        resource_methods = self.context.resource["resourceMethods"]
+        if self.context.method not in resource_methods:
+            return True
+
+        # check if there is validator for the resource
+        resource = resource_methods[self.context.method]
+        if (
+            "requestValidatorId" not in resource
+            or resource["requestValidatorId"] is None
+            or not resource["requestValidatorId"].strip()
+        ):
+            return True
+
+        # check if there is a validator for this request
+        validator = self.apigateway_client.get_request_validator(
```

Comment:
Since we're only using the `apigateway_client` here, and probably a large portion of user requests in API GW do not use a request validator, I think it would be advisable to instantiate the `apigateway_client` directly here (could improve performance a bit, as boto3 client creation may come with an overhead).

Reply:
It makes testing a bit more difficult. I do agree with the overhead of course, but one can cache since is used in different places during the request